ddd-kata.md

```diff
@@ -0,0 +1,197 @@
+
+## The DDD Kata
+
+### Task
+
+* Select a collaboration tool for your choice. In SAP we are using Mural. In the DDD Community Miro is used a lot.
+* **Form a working group** (up to 8 people). Declare one member a domain expert (maybe a substitute) and one member as facilitator / moderator
+* Perform a **Big Picture Event Storming** with your group in collaboration tool of choice
+* **Identify the bounded context and aggregates** within evaluated process in collaboration tool of choice
+* Establish the message flow from Bounded Context to Bounded Context using the **Domain Message Flow** Diagrams in collaboration tool of choice.
+* For each identified Bounded Context, specify using the **Bounded Context Canvas** in collaboration tool of choice.
+* Within the defined Bounded Contexts, defined the aggregates using the **Aggregate canvases** in collaboration tool of choice.
+
+### The Requirements
+
+Let’s assume you are a start-up, and you want to provide software for parking lots & garages for all major cities in Europe and for various park area providers.
+
+1. You can enter and leave the parking lot or the parking garage through several entrances and exits.
+2. There are specific parking spots for different types of vehicles (motorcycles, car, electric car, truck/bus, handicapped persons, family-friendly parking spots)
+3. Each parking spot has a unique identifier (for parking garage – number & floor or for parking lot – number &  area)
+4. The customer may collect a ticket at each entrance and leave via any exit.
+5. The ticket given to customer contains a parking spot ID where to park
+6. The ticketing information must be stored for 10 years due to tax reasons.
+7. There are three types of terminals:
+   1. The terminals at the entrances have a way for the customer to communicate the vehicle details to the system.
+   2. Customer drive into the parking garage / lot via entrance terminals where the collect the tickets
+   3. The terminals near the exits have a way to pay the ticket.
+   4. Detail out the actual payment process like select payment method, put in credit card, enter PIN are not relevant here.
+8. The terminals at the exit gates collect the paid ticket.
+9. The ticket has a read/write magnetic stripe containing information including a position to park.
+10. The magnetic stripe can be changed by the terminals.
+11. Parking garages have a parking guiding system, that guides customers to the correct parking spot. The guiding system recognizes the vehicle and guides it to the right parking spot.
+12. Parking lots do not have such a system.
+
+As domain expert, please stay from requirements perspective within these guardrails. You can detail them out as much as you want.
```

Comment:
```suggestion
As domain expert, please stay within these guardrails when detailing out the requirements, if possible.
In some cases, it can make sense to challenge/rethink the original requirements, but be careful to not modify them in such a way that the inherent complexity becomes too limited.
The focus of the kata is learning DDD, not sticking to the requirements word for word.
However, some amount of complexity is important to get a realistic impression.
```

Reply:
@neolytian FYI, I added some content here